src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -602,7 +608,7 @@ Status LDAPManagerImpl::execQuery(std::string& ldapurl, std::vector<std::string>
     while (entry) {
         if (entitiesonly) {
             auto dn = ldap_get_dn(ldap, entry);
-            ON_BLOCK_EXIT([&] { ldap_memfree(dn); });
+            ON_BLOCK_EXIT([=] { ldap_memfree(dn); });
```

Comment:
As far as I can see, this lambda is going to be executed after the `dn` pointer has been placed into the `results` vector in line 619. Does that mean that `results` will effectively keep a dangling pointer?

Reply:
No, std::string constructor at line 619 will allocate memory and copy string from `dn`.